### ISSUE TITLE
Allow overriding of  docker volume argument

### ISSUE THREAD

**Issue by Kenchir** [pypa/cibuildwheel#1669]
### Description

The current ci_buildwheel runs a docker container with `/` mounted  to '/host'. volume This can perhaps be made dynamic so that it can be overridden  when passed as argument.

**Comment by joerick** [comment]
Hi. We're unlikely to add a feature without understanding the problem. Can I ask why you'd like that?

**Comment by Kenchir** [comment]
Hi @joerick , I have edited  my issue. So problem happens when the root volume is restricted on the case of  [bitbucket](https://support.atlassian.com/bitbucket-cloud/docs/run-docker-commands-in-bitbucket-pipelines/#:~:text=docker%20run%20%2D%2Dmount-,Full%20list%20of%20restricted%20commands,%2D%2Dsecurity%2Dopt,-Using%20Docker%20Compose) I would like to pass the volume argument and the default one  gets overridden.

**Comment by joerick** [comment]
Does it error out currently with the use of `--mount`? Or just fail silently?

**Comment by Kenchir** [comment]
From the. Bitbucket  docs, `--mount` is not allowed leaving `--volume`  as the only option. Below is the error.

```
Starting container image kenchir/manylinux_x86_64...
info: This container will host the build for cp39-manylinux_x86_64, cp311-manylinux_x86_64...
Error response from daemon: authorization denied by plugin pipelines: -v only supports $BITBUCKET_CLONE_DIR and its subdirectories
                                                                       ✕ 0.03s
Error: Command ['docker', 'create', '--env=CIBUILDWHEEL', '--env=SOURCE_DATE_EPOCH', '--name=cibuildwheel-e6aedf54-1315-4495-9423-484147dc9662', '--interactive', '--volume=/:/host', '--cpus=3.5', '-m=10G', '--volume=/opt/atlassian/pipelines/agent/build/python-dtone-cdrparser:/host', 'kenchir/manylinux_x86_64', '/bin/bash'] failed with code 1. None
```

As you can see, it tries to `/`  is disabled for security purpose.

**Comment by joerick** [comment]
I can imagine other reasons why users might want to disable that (isolation), and luckily this was quite easy to add, so I've pushed a PR.

**Comment by Kenchir** [comment]
Awesome!

**Comment by joerick** [comment]
sorry for the delay, i rather forgot about this! Just released as v2.16.3.
